detection_rules/rule.py

```diff
@@ -1334,6 +1341,22 @@ def save_json(self, path: Path, include_version: bool = True):
             json.dump(self.contents.to_api_format(include_version=include_version), f, sort_keys=True, indent=2)
             f.write('\n')
 
+    def validate_version_revision(self):
+        """Validates 'version' and 'revision' fields based on the rule directory."""
+
+        if self.path and (RULES_DIR in self.path.parents or RULES_BBR_DIR in self.path.parents):
```

Comment:
I think a simpler way would be to check if the custom dir envvar is set

```python
if not os.environ.get('CUSTOM_RULES_DIR')
```

Then I would raise `ValidationError`